### PR TITLE
Every observer now has persistent HUDs defined by the client preferences

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

**Comment by Ghommie** [comment]
It's preferable for this to be an option rather something mandatory on new ghosts/observers, though I do dislike having to manually set each one.

**Comment by DominikPanic** [comment]
Yep, I'll look into player preferences to make this more better, this is just a temporary solution.

**Comment by DominikPanic** [comment]
Do not merge until I figure a very bad bug out.

**Comment by DominikPanic** [comment]
Alright it should work now (hopefully)

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -228,33 +253,77 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+	ghost_sechud = client.prefs.ghost_sechud
+	ghost_squadhud = client.prefs.ghost_squadhud
+	ghost_xenohud = client.prefs.ghost_xenohud
 	var/datum/mob_hud/H
-	var/HUD_nbr = 1
+
 	switch(hud_choice)
 		if("Medical HUD")
-			H = huds[MOB_HUD_MEDICAL_OBSERVER]
+			if(ghost_medhud)
```
Comment (Rohesie) [inline code/modules/mob/dead/observer/observer.dm]:
```suggestion
			ghost_medhud = !ghost_medhud
			H = huds[MOB_HUD_MEDICAL_OBSERVER]
			ghost_medhud ? H.add_hud_to(src) : H.remove_hud_from(src)
			client.prefs.ghost_medhud = ghost_medhud
			client.prefs.save_preferences()
			to_chat(src, (ghost_medhud ? "\blue <B>[hud_choice] Enabled</B>" : "\blue <B>[hud_choice] Disabled</B>"))
```
This kind of logic can condense the code.

**Pull request merged**